### PR TITLE
Fileio/read

### PR DESCRIPTION
**Pull request by WaelYoussfi**
starting the read intrinsic implementation in the c backend.
it still needs the file_size intrinsic to work and that will be fixed in the next pr

### CODE REVIEW

File: src/dev/flang/be/c/C.java [hunk 1]
```diff
@@ -605,6 +605,26 @@ private void createCode(CFile cf)
...
+    // start of fileio related intrinsics
+    // fuzion.std.fileio.read
+    // currently logs console messages instead of proper returns in case of errors
+    cf.print
+      (CStmnt.lineComment("helper to read bytes from a file"));
+      cf.print
+        ("void "+CNames.FILEIO_READ._name+"(char* path, void* content, size_t size)\n"+
+          "{FILE *f = fopen(path, \"r\");\n"+
+          "if (f == NULL)\n"+
+          "{printf(\"error happened\");\n"+
```
Comment (fridis) [inline src/dev/flang/be/c/C.java:617]: please add a proper error message. Where did it happen? what happened? Also, please print error to stderr. Start the printed message with `*** `.

File: src/dev/flang/be/c/C.java [hunk 2]
```diff
@@ -605,6 +605,26 @@ private void createCode(CFile cf)
...
 
+    // start of fileio related intrinsics
+    // fuzion.std.fileio.read
+    // currently logs console messages instead of proper returns in case of errors
+    cf.print
+      (CStmnt.lineComment("helper to read bytes from a file"));
+      cf.print
+        ("void "+CNames.FILEIO_READ._name+"(char* path, void* content, size_t size)\n"+
+          "{FILE *f = fopen(path, \"r\");\n"+
+          "if (f == NULL)\n"+
```
Comment (fridis) [inline src/dev/flang/be/c/C.java:616]: please use indentation in the generated C code.

File: src/dev/flang/be/c/C.java [hunk 3]
```diff
@@ -605,6 +605,26 @@ private void createCode(CFile cf)
...
+          "{printf(\"error happened\");\n"+
+          "exit(0);}\n"+
+          "else {\n"+"size_t reading = fread(content, 1, size, f);\n"+
+          "if (reading == 0)\n"+
+          "printf(\"there is no more bytes to read\");\n"+
```
Comment (fridis) [inline src/dev/flang/be/c/C.java:621]: please use { } to enclose code blocks after `if`, `else`, etc.

The `printf` here looks like early debug output. Please mark with a comment `// NYI: <reason>` and add a reason instead like `under development, need to remove this debug output` or similar.

File: src/dev/flang/fuir/analysis/dfa/DFA.java [hunk 4]
```diff
@@ -915,7 +915,7 @@ static Value NYIintrinsicMissing(Call cl)
     put("fuzion.std.exit"                , cl -> null );
     put("fuzion.std.out.write"           , cl -> Value.UNIT );
     put("fuzion.std.err.write"           , cl -> Value.UNIT );
-    put("fuzion.std.fileio.read"         , cl -> NYIintrinsicMissing(cl) );
+    put("fuzion.std.fileio.read"         , cl -> Value.UNIT );
```
Comment (fridis) [inline src/dev/flang/fuir/analysis/dfa/DFA.java:918]: please add a comment `NYI: need to record modification of array passed as argument` or similar.

**Review by fridis — COMMENTED** [review]
Looks good. Some minor comments related to debug output, formatting and marking code with `NYI`.

File: src/dev/flang/be/c/C.java (hunk 3 above)
Comment (WaelYoussfi) [reply src/dev/flang/be/c/C.java:621]: should the comment be in the generated c file or in C.java?

File: src/dev/flang/be/c/C.java [hunk 5]
```diff
@@ -605,6 +605,40 @@ private void createCode(CFile cf)
...
+    cf.print
+      (CStmnt.lineComment("helper to read bytes from a file"));
+      cf.print
+        ("void "+CNames.FILEIO_READ._name+"""
+        (char* path, void* content, size_t size)
+        {
+          FILE *f = fopen(path, \"r\");
+          if (f == NULL)
+          {
+            fprintf(stderr, \"*** failed to open the file %s \\n\", path);
```
Comment (fridis) [inline src/dev/flang/be/c/C.java:620]: Please add a comment `/* NYI: proper error handling mising */`

File: src/dev/flang/be/c/C.java [hunk 6]
```diff
@@ -605,6 +605,40 @@ private void createCode(CFile cf)
...
+            exit(0);
+          }
+          else 
+          {
+            size_t reading = fread(content, 1, size, f);
+            if (reading != size)
+            {
+              if (ferror(f)!=0)
+              {
+                fprintf(stderr, \"*** failed to read the file %s \\n\", path);
```
Comment (fridis) [inline src/dev/flang/be/c/C.java:630]: Please add a comment `/* NYI: proper error handling mising */`

File: src/dev/flang/be/c/C.java [hunk 7]
```diff
@@ -605,6 +605,40 @@ private void createCode(CFile cf)
...
+              }
+              else if (feof(f)!=0)
+              {
+                fprintf(stdout, \"reached end of the file %s \\n\", path);
```
Comment (fridis) [inline src/dev/flang/be/c/C.java:634]: Please add a comment describing why this code is here, e.g.,  `/* NYI: remove debug output */`

**Pull request merged**